ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/AdminLogsRepairCache.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.ethereum.api.query.BlockchainQueries;
+import org.hyperledger.besu.ethereum.api.query.TransactionLogBloomCacher;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class AdminLogsRepairCache implements JsonRpcMethod {
+  private final BlockchainQueries blockchainQueries;
+
+  public AdminLogsRepairCache(final BlockchainQueries blockchainQueries) {
+    this.blockchainQueries = blockchainQueries;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ADMIN_LOGS_REPAIR_CACHE.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    final Optional<Long> blockNumber = requestContext.getOptionalParameter(0, Long.class);
+
+    blockNumber.ifPresent(
+        bn ->
+            blockchainQueries
+                .getBlockchain()
+                .getBlockByNumber(bn)
+                .orElseThrow(() -> new IllegalStateException("Block not found, " + bn)));
+
+    final TransactionLogBloomCacher transactionLogBloomCacher =
+        blockchainQueries
+            .getTransactionLogBloomCacher()
+            .orElseThrow(
+                () -> new InternalError("Error attempting to get TransactionLogBloomCacher"));
```

Comment:
We should make this actionable since this is inscrutable to anyone who doesn't know our architecture. If there isn't a `TransactionLogBloomCacher`, that means the associated option must be disabled. We should tell them that this rpc method doesn't do anything unless they have the cacher enabled.

Reply:
I can update this message, but don't follow what you mean by actionable. I will update the message to state more info about the need for the TransactionLogBloomCacher but aside from that I don't think there is much more this end point should do for them.

Reply:
It's the difference between the user seeing "Error something didn't work...", which might make them think it's our problem, vs "Can't repair caches because the transaction log bloom cacher isn't enabled. Try enabling it with `insert option name here`"

Reply:
Basically what I mean by actionable is that it empowers them to understand and fix the issue since the reason they're trying to access this method is because things are going wrong.

I can easily imagine this happening:
logs are innacurate -> they turn log caching off -> they forget to turn it back on -> they try to repair it -> we give them another error that they don't understand

We want to change that last step so it's instead:
"oh, I didn't know I needed to enable the caching going forward to be able to repair it, I'll go do that now."

Reply:
I will make the update but this is possibly our issue. You need to understand that when something like this goes wrong it is not always just the user entering the wrong command, it can also be a major issue within the software itself.

Reply:
You're right that there's one other case. If we had an empty optional even though the user had the log cacher enabled that means that we have a bug in our configuration code. Even this eventuality can be handled with a sentence like "If this message persists, create an issue."

I guess what I'm having a hard time understanding is what is the argument _for_ terse and opaque error messages. We don't have word limits on these. As we improve the software and have more people relying on us, we want to create as painless of an experience for them as possible. Like I said before working with this endpoint at all is a result of instability in the log cacher, so things are already going wrong. I'm on linux so I know how it feels to hit error after error in software someone else wrote ;) It's pretty frustrating.

Reply:
Agreed, my initial message was really not helpful. I think this is something we can improve as you mention as we go forward.